File: ARCADE/Intro/Level 10_4: electionsWinners.cs
    using System;
     
    public class Test
    {
    	public static int electionsWinners(int[] votes, int k) {
    	    int count = 0;      
    	    Array.Sort(votes);
    	    int max = votes[votes.Length - 1];
    	    if(k == 0){
    	        for(int i = 0; i < votes.Length;i++){
    	            if(votes[i] == max) count++;
    	        }
    	        if(count >= 2) return 0;
    	        if(count == 1) return 1;
    	    }
    	    else{
    	        for(int i = 0; i < votes.Length;i++){
    	            if(votes[i] + k > max) count++;
    	        }
    	    }
    	    return count;
    	}
     
    	public static void Main()
    	{
    		 int [] votes = new int []{2, 3, 5, 2};
    		 int k = 3;
    		 Console.Write(electionsWinners(votes, k));
    	}
    }

File: ARCADE/Intro/Level 10_4: electionsWinners.pas
program electionsWinners;
uses Math;
Type
	ArrayInteger = Array Of LongInt;

function electionsWinners(votes: ArrayInteger; k: integer): integer;
var i, rs, max, count : integer;
begin
    rs := 0; count := 0;
    max := MaxValue(votes);
    if(k = 0) then 
    begin
        for i := 0 to length(votes) - 1 do
            if(votes[i] = max) then inc(count);
        if(count > 1) then electionsWinners := 0
        else electionsWinners := 1;
        exit;
    end;
    
    max := MaxValue(votes);
    for i := 0 to length(votes) - 1 do
        votes[i] := votes[i] + k;
        
    for i := 0 to length(votes) - 1 do
        if(votes[i] > max) then inc(rs);
    
    electionsWinners := rs;
end;

var votes : ArrayInteger;
	k : integer;
begin
	votes := ArrayInteger.Create(2, 3, 5, 2);
	k := 3;
	writeln(electionsWinners(votes, k));
end.
